py/vm.c

```diff
@@ -1256,12 +1256,27 @@ unwind_jump:;
 
 pending_exception_check:
                 MICROPY_VM_HOOK_LOOP
+                #if MICROPY_PY_SOFTIRQ
+                if (MP_STATE_VM(mp_pending_ex_flags)) {
```

Comment:
From this passage it's clear that there's no need to introduce mp_pending_ex_flags - we already have MP_OBJ_SENTINEL which can be stored in any mp_obj_t.